compiler-hacking.md

```diff
@@ -0,0 +1,474 @@
+# Understanding the Shemsu compiler
+This is a guide to understanding the Shesmu olive compiler. This is meant to
+explain the core components that get the language to Java bytecode.
+
+It assumes you have a working understanding of Java bytecode. The [DZone
+bytecode
+tutorial](https://dzone.com/articles/nominalized-adjectives-as-names-for-decorators)
```

Comment:
Is this link correct?

Reply:
No! And yet it is also the link in the implementation guide. :cry: